packages/firestore/src/core/sync_engine.ts

```diff
@@ -770,29 +781,44 @@ export class SyncEngine implements RemoteSyncer, SharedClientStateSyncer {
     const key = limboChange.key;
     if (!this.limboTargetsByKey.get(key)) {
       log.debug(LOG_TAG, 'New document in limbo: ' + key);
+      this.limboListenQueue.push(key);
+      this.startEnqueuedLimboResolutions();
+    }
+  }
+
+  private startEnqueuedLimboResolutions(): void {
```

Comment:
I will add the requested documentation to the method.

Regarding the naming issue... what about using the verb "trigger" in the method name?  The "pipeline" idea makes sense too, but the method in question is not technically "filling" the pipeline and I find that verb to be misleading.  The "pump" verb is sounding good to me though.  According to https://en.wikipedia.org/wiki/Event_loop the verb "pump" has a meaning in an event loop/message queue context, which has many analogues to the limbo resolution queue.  That article also suggests the verb "dispatch".

So here are some other names to consider:
- `triggerLimboResolutions()`
- `pumpLimboResolutionQueue()` (my preference)
- `dispatchLimboResolutionListens()`

Regarding the limbo resolution state names, I'm leaning towards the terms "enqueued" and "active".  I prefer "enqueued" over "pending" because, IMO, it is a more concrete term than "pending" and therefore requires less explanation in documentation.

Reply:
What you implemented was `pumpLimboResolutionListenQueue`. I offer that you should construct this as "pump" plus the name of the thing. If you go with `enqueuedLimboResolutions`, above, then match with `pumpEnqueuedLimboResolutions`. 

I'm also fine with "trigger" if that reads better to you.

Reply:
Done.  I've renamed the method to `pumpEnqueuedLimboResolutions()`.